src/audio/kpb.c

```diff
@@ -139,12 +139,15 @@ static struct comp_dev *kpb_new(struct sof_ipc_comp *comp)
 		return NULL;
 	}
 
-	dev->state = COMP_STATE_READY;
-	kpb->history_buffer = NULL;
 
-	/* Zero number of clients */
+	/* Init basic component data */
+	kpb->history_buffer = NULL;
 	kpb->kpb_no_of_clients = 0;
 
+	/* Kpb has been created successfully */
+	dev->state = COMP_STATE_READY;
+	kpb->state = KPB_STATE_CREATED;
```

Comment:
Do we have a state transition diagram like we do for generic components ? This is needed given the number of states we have in KPB.

Reply:
@lgirdwood, I am afraid we don't have such a diagram. These states are KPB internal, no other component or pipeline should be aware of it. The main purpose of these quite detailed states is to decide if we can reset/free KPB right away or should be postpone after we gracefully terminate current KPB task - draining or buffering states are a good example here. Secondly, since KPB is a heart of KWD its good to have such detailed states in regards to debugging - I am going to add a new patch which records last eight states of KPB. So every time something wrong happened we can check how KPB was changing its states.